### PR TITLE
Mid term project mihaela si bianca

### PR DESCRIPTION
**Pull request by MihaelaGit**

### CODE REVIEW

File: midterm_project/project_mihaela_bianca/shopping_list.py [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+        {egg:3, milk:3, flour:2, sugar:1}
+        """
+        self.name = name
+        self.ingredients = ingredients
+
+        if len(self.ingredients) < 4:
+            raise Exception('The ingredients number should be  at least 4 !')
+
+    def __iter__(self):
+        yield from self.ingredients.items()
```
Comment (Liitle) [inline midterm_project/project_mihaela_bianca/shopping_list.py]: nice use of yield. An alternative here could be `iter(self.ingredients.items())`

File: midterm_project/project_mihaela_bianca/shopping_list.py [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+
+    def __init__(self, *args):
+        """
+         RecipesBox should be a list that holds our recipes
+         [...]
+        """
+        self.recipes = list(args)
+
+    def __iter__(self):
+        yield from self.recipes
```
Comment (Liitle) [inline midterm_project/project_mihaela_bianca/shopping_list.py]: same here, `iter(self.recipes) `can also work

**Review by Liitle — APPROVED** [review]
Good Job!

File: midterm_project/project_mihaela_bianca/shopping_list.py (hunk 2 above)
Comment (MihaelaGit) [reply midterm_project/project_mihaela_bianca/shopping_list.py]: modified

File: midterm_project/project_mihaela_bianca/shopping_list.py (hunk 1 above)
Comment (MihaelaGit) [reply midterm_project/project_mihaela_bianca/shopping_list.py]: modified

**Pull request merged**